### PR TITLE
Revamp lima website using docsy

### LINKED ISSUE #1499: Web site (https://lima-vm.io/): migrate to docsy, mkdocs, gitbook, or something else
The https://lima-vm.io/ website just (poorly) renders markdown on the client side.
We should consider migrating the website to docsy, mkdocs, gitbook, or something else.

## Requirements
- The site SHOULD be multi-versioned (v0.15.x, v0.16.x, master, ...)
- The site MAY support multiple languages, but its maintenance cost will be probably high. Just linking to a machine translation site might be enough.

- The site MUST have the `limactl` CLI reference
- The site MUST have the built-in template list. Just linking to https://github.com/lima-vm/lima/blob/master/examples/README.md might be enough.
- The site SHOULD have the YAML reference

## Resources
Netlify base directory: https://github.com/lima-vm/lima/tree/master/website

### PR DESCRIPTION
**Pull request by balajiv113**
Fixes #1499 

Creating as draft to keep track of the progress

- [X] Home page
- [x] Documentation - Installation
- [x] ~~Documentation - Getting Started~~
- [x] Documentation - Templates
- [x] ~~Documentation - Architecture and other informations present under docs folder~~
- [x] Documentation - Cli reference
- [x] Contribution
- [x] FAQ

### CODE REVIEW

File: website/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+We'd love to accept your patches and contributions to this project. There are
+just a few small guidelines you need to follow.
+
+## Contributor License Agreement
+
+Contributions to this project must be accompanied by a Contributor License
+Agreement. You (or your employer) retain the copyright to your contribution;
+this simply gives us permission to use and redistribute your contributions as
+part of the project. Head over to <https://cla.developers.google.com/> to see
+your current agreements on file or to sign a new one.
```
Comment (AkihiroSuda) [inline website/CONTRIBUTING.md:12]: ?

File: website/README.md [hunk 2]
```diff
@@ -1,12 +1,154 @@
-# The source of the Lima website (https://lima-vm.io)
+# Docsy Example
```
Comment (AkihiroSuda) [inline website/README.md:1]: Docsy shouldn’t be pushed to this repo, and it should be pulled from the upstream in netlify.toll

File: website/CONTRIBUTING.md (hunk 1 above)
Comment (balajiv113) [reply website/CONTRIBUTING.md:12]: These are pre generated files. Will remove this

File: website/data/adoptors.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+- name: Rancher Desktop
+  link: https://rancherdesktop.io
+  image: rancherdesktop.svg
+  width: 200
+- name: Colima
+  link: https://github.com/abiosoft/colima
+  image: colima.png
```
Comment (AkihiroSuda) [inline website/data/adoptors.yaml:7]: - https://github.com/abiosoft/colima/issues/781

File: website/README.md (hunk 2 above)
Comment (balajiv113) [reply website/README.md:1]: The heading is wrong. Need to update readme for website code. Will do that 

Docsy is fetched as a module only. Its not pushed here

File: website/hugo.toml [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+  desc = "Discuss development issues around the project"
+
+[module]
+  # uncomment line below for temporary local development of module
+  # replacements = "github.com/google/docsy -> ../../docsy"
+  [module.hugoVersion]
+    extended = true
+    min = "0.110.0"
+  [[module.imports]]
+    path = "github.com/google/docsy"
```
Comment (balajiv113) [inline website/hugo.toml:184]: FYI - This is where docsy is imported.

**Comment by balajiv113** [comment]
@AkihiroSuda 
In recent changes i automated generation of markdown for docsy using cobra api. This way the cli-references will always be upto date.

Hope this is fine. Let me know if you some feedbacks.

As next steps will update examples & contribution guidelines.

**Comment by afbjorklund** [comment]
> In recent changes i automated generation of markdown for docsy using cobra api. This way the cli-references will always be upto date.

Why do this in _this_ PR, and why in the "gen-man" command ? There was already a "gen-doc" on the way...

**Comment by balajiv113** [comment]
> Why do that in this PR

Will put down a separate PR if this approach is fine. 

> and why in the "gen-man" command

Ah yes my bad, gen-man should not be touched for this. Maybe will add a gen-doc like you have done before. But the markdown generated looks somewhat acceptable only.

**Comment by balajiv113** [comment]
This PR is ready. I have mostly covered all sections and also fixed relative links in a dynamic fashion to avoid too many duplicates.

Feel free to put down ideas/comments on contents as well as look and feel.

We can review and merge this after https://github.com/lima-vm/lima/pull/1778 is merged

**Comment by AkihiroSuda** [comment]
Please update [website/CONTRIBUTING.md](https://github.com/lima-vm/lima/pull/1754/files#diff-ccf281dd7e5b2b0bfa1fb39f5ab55513315b70dde639fe4428ff064ce6c53e33) and similar files

**Comment by balajiv113** [comment]
> Please update [website/CONTRIBUTING.md](https://github.com/lima-vm/lima/pull/1754/files#diff-ccf281dd7e5b2b0bfa1fb39f5ab55513315b70dde639fe4428ff064ce6c53e33) and similar files

I have removed those files from website folder for now. I will raise a separate PR for contributing to website.

File: website/README.md [hunk 5]
```diff
@@ -1,12 +0,0 @@
-# The source of the Lima website (https://lima-vm.io)
-
-This directory is the [Netlify base directory](https://docs.netlify.com/configure-builds/overview/) of [https://lima-vm.io](https://lima-vm.io/) .
```
Comment (AkihiroSuda) [inline website/README.md:3]: Can we keep this?

File: website/layouts/partials/footer.html [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        {{- end -}}
+        {{ end }}
+      </div>
+    </div>
+    <div class="row text-center text-white small">
+      <div id="cncf-footer" class="col-12 text-center py-2 order-sm-2">
+        <a class="text-secondary" href="https://www.linuxfoundation.org/terms" target="_blank" rel="noopener">Terms</a> |
+        <a class="text-secondary" href="https://www.linuxfoundation.org/privacy" target="_blank" rel="noopener">Privacy</a> |
+        <a class="text-secondary" href="https://www.linuxfoundation.org/trademark-usage" target="_blank" rel="noopener">Trademarks</a> |
+        <a class="text-secondary" href="https://github.com/grpc/grpc.io/blob/main/LICENSE" target="_blank" rel="noopener">License</a>
```
Comment (AkihiroSuda) [inline website/layouts/partials/footer.html:38]: No grpc

Comment (balajiv113) [reply website/layouts/partials/footer.html]: Done

File: website/README.md (hunk 5 above)
Comment (balajiv113) [reply website/README.md]: Retained

File: website/README.md [hunk 7]
```diff
@@ -9,4 +9,4 @@ The actual contents are generated from the markdown files on the browser side:
 
 The site is previewable and deployable with just the single [`index.html`](./index.html).
 
-No dependency on any templating engine currently, but eventually we may adopt docsy or something else similar.
+No dependency on any templating engine currently, but eventually we may adopt docsy or something else similar.
```
Comment (AkihiroSuda) [inline website/README.md:12]: I meant to keep L1-L3.
Other lines have to be removed

File: website/go.mod [hunk 8]
```diff
@@ -0,0 +1,9 @@
+module github.com/google/docsy-example
```
Comment (AkihiroSuda) [inline website/go.mod:1]: ?

File: website/go.mod [hunk 9]
```diff
@@ -0,0 +1,9 @@
...
+
+go 1.12
```
Comment (AkihiroSuda) [inline website/go.mod:3]: outdated

File: website/go.mod
Comment (AkihiroSuda) [inline website/go.mod:1]: Why do we need this?

Comment (balajiv113) [reply website/go.mod:1]: Without this file it fails to start

File: website/README.md (hunk 7 above)
Comment (balajiv113) [reply website/README.md]: My bad. Done the changes

File: website/go.mod (hunk 8 above)
Comment (balajiv113) [reply website/go.mod]: Replaced

File: website/go.mod (hunk 9 above)
Comment (balajiv113) [reply website/go.mod]: Updated

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Comment by AkihiroSuda** [comment]
@lima-vm/maintainers 

I'm merging this, but please do review this and open follow-up issues and PRs.

**Comment by afbjorklund** [comment]
What is with the Title Case ?

```console
$ ls website/content/en/docs/
Examples  faq  _index.md  Installation  Reference  Templates  Usage
```

Probably a Mac thing.

**Pull request merged**